client/js/lounge.js

```diff
@@ -74,28 +74,30 @@ $(function() {
 		});
 	});
 
-	socket.on("auth", function(/* data */) {
+	socket.on("auth", function(data) {
 		var body = $("body");
 		var login = $("#sign-in");
 		if (!login.length) {
 			refresh();
 			return;
 		}
+
 		login.find(".btn").prop("disabled", false);
-		var token = window.localStorage.getItem("token");
-		if (token) {
-			window.localStorage.removeItem("token");
```

Comment:
This is the actual offender. I chose to never delete the token even on failed auths to show the "authorisation failed" message to the user until they relogin.